### PR TITLE
test(core): oa integration tests

### PR DESCRIPTION
**Pull request by Reinis-FRP**
Signed-off-by: Reinis Martinsons <reinis@umaproject.org>

**Motivation**

Verify Optimistic Asserter interacts with Escalation Managers as expected.

**Summary**

Adds integration tests to Optimistic Asserter paired with Base Escalation Manager and Full Policy Escalation Manager.

**Details**

This also adds `OptimisticAsserterTest` as a wrapper contract to pass through assertions on user's behalf. This is required to properly distinguish between asserter and asserting caller in the integration tests.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/core/test/foundry/optimistic-asserter/Common.sol [hunk 1]
```diff
@@ -147,6 +158,24 @@ contract Common is Test {
             );
     }
 
+    function _wrappedAssertWithCallbackRecipientAndSs(address callbackRecipient, address escalationManager)
```
Comment (Reinis-FRP) [inline packages/core/test/foundry/optimistic-asserter/Common.sol:161]: The naming should be updated to `_wrappedAssertWithCallbackRecipientAndEm`, but plan to do it in separate PR as there are multiple references to SS across the tests.

File: packages/core/test/foundry/optimistic-asserter/integration-tests/BaseEscalationManager.t.sol [hunk 2]
```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "../Common.sol";
+import "../../../../contracts/optimistic-asserter/implementation/escalation-manager/BaseEscalationManager.sol";
+
+contract BaseEscalationManagerTest is Common {
```
Comment (chrismaree) [inline packages/core/test/foundry/optimistic-asserter/integration-tests/BaseEscalationManager.t.sol:7]: these are awesome!

**Review by chrismaree — APPROVED** [review]
